src/api/app/mixins/webui/has_flags.rb

```diff
@@ -92,4 +92,45 @@ def get_build_flags
     return build_flags
   end
 
+  def existing_flag(options = {})
+    main_object.flags.where(
+      repo: options[:repo], architecture: options[:architecture]
+    ).first
+  end
+
+  def temporary_flag(attributes = {})
+    attributes.update(flag: 'build') # make it configurable
+    attributes[:status] = Flag.default_state('build') unless attributes[:status]
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.